File: reference/SmartPharmacy-Kotlin/app/src/main/java/com/example/ui/screens/ExpiryScreen.kt
package com.example.ui.screens

import android.widget.Toast
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material.icons.filled.Discount
import androidx.compose.material.icons.filled.Warning
import androidx.compose.material3.*
import androidx.compose.material3.TabRowDefaults.tabIndicatorOffset
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.ui.components.MedicineCard
import com.example.ui.theme.*
import com.example.ui.viewmodel.PharmacyViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ExpiryScreen(
    viewModel: PharmacyViewModel,
    onNavigateToProductDetails: (String) -> Unit,
    onNavigateBack: () -> Unit
) {
    val context = LocalContext.current
    val medicines by viewModel.medicines.collectAsState()

    val expiredList = remember(medicines) { viewModel.getExpiredMedicines(medicines) }
    val expiring30DaysList = remember(medicines) { viewModel.getExpiringIn30DaysMedicines(medicines) }

    var selectedTab by remember { mutableStateOf(0) } // 0: Expired, 1: Expiring Soon

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Expiry Management", fontWeight = FontWeight.Bold) },
                navigationIcon = {
                    IconButton(onClick = onNavigateBack) {
                        Icon(imageVector = Icons.Default.ArrowBack, contentDescription = "Back")
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = SurfaceWhite)
            )
        },
        containerColor = SurfaceWhite
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding)
                .padding(horizontal = 16.dp)
        ) {
            Spacer(modifier = Modifier.height(12.dp))

            // Tab Selector
            TabRow(
                selectedTabIndex = selectedTab,
                containerColor = SurfaceCard,
                contentColor = PrimaryBlue,
                indicator = { tabPositions ->
                    TabRowDefaults.SecondaryIndicator(
                        modifier = Modifier.tabIndicatorOffset(tabPositions[selectedTab]),
                        color = if (selectedTab == 0) AlertRed else WarningOrange
                    )
                }
            ) {
                Tab(
                    selected = selectedTab == 0,
                    onClick = { selectedTab = 0 },
                    text = {
                        Text(
                            text = "Expired (${expiredList.size})",
                            fontWeight = FontWeight.Bold,
                            color = if (selectedTab == 0) AlertRed else TextSecondary
                        )
                    }
                )

                Tab(
                    selected = selectedTab == 1,
                    onClick = { selectedTab = 1 },
                    text = {
                        Text(
                            text = "Expiring in 30 Days (${expiring30DaysList.size})",
                            fontWeight = FontWeight.Bold,
                            color = if (selectedTab == 1) WarningOrange else TextSecondary
                        )
                    }
                )
            }

            Spacer(modifier = Modifier.height(16.dp))

            val currentDisplayList = if (selectedTab == 0) expiredList else expiring30DaysList

            if (currentDisplayList.isEmpty()) {
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .weight(1f),
                    contentAlignment = Alignment.Center
                ) {
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Icon(
                            imageVector = Icons.Default.Warning,
                            contentDescription = null,
                            tint = SuccessGreen,
                            modifier = Modifier.size(48.dp)
                        )
                        Spacer(modifier = Modifier.height(12.dp))
                        Text(
                            text = if (selectedTab == 0) "No Expired Medicines!" else "No Medicines Expiring in 30 Days!",
                            fontSize = 16.sp,
                            fontWeight = FontWeight.Bold,
                            color = TextPrimary
                        )
                        Text(
                            text = "Your inventory is fresh and safe.",
                            fontSize = 13.sp,
                            color = TextSecondary
                        )
                    }
                }
            } else {
                LazyColumn(
                    verticalArrangement = Arrangement.spacedBy(12.dp),
                    contentPadding = PaddingValues(bottom = 24.dp)
                ) {
                    items(currentDisplayList, key = { it.id }) { medicine ->
                        Column {
                            MedicineCard(
                                medicine = medicine,
                                isExpired = selectedTab == 0,
                                isExpiringSoon = selectedTab == 1,
                                onClick = { onNavigateToProductDetails(medicine.id) }
                            )

                            Spacer(modifier = Modifier.height(6.dp))

                            Row(
                                modifier = Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.End
                            ) {
                                if (selectedTab == 1) {
                                    TextButton(
                                        onClick = {
                                            Toast.makeText(context, "Clearance 30% discount tag added to ${medicine.name}", Toast.LENGTH_SHORT).show()
                                        }
                                    ) {
                                        Icon(imageVector = Icons.Default.Discount, contentDescription = null, tint = WarningOrange, modifier = Modifier.size(16.dp))
                                        Spacer(modifier = Modifier.width(4.dp))
                                        Text("Apply Clearance Discount", fontSize = 12.sp, color = WarningOrange, fontWeight = FontWeight.Bold)
                                    }
                                }

                                TextButton(
                                    onClick = {
                                        viewModel.deleteMedicine(medicine.id)
                                        Toast.makeText(context, "${medicine.name} removed from inventory", Toast.LENGTH_SHORT).show()
                                    }
                                ) {
                                    Icon(imageVector = Icons.Default.Delete, contentDescription = null, tint = AlertRed, modifier = Modifier.size(16.dp))
                                    Spacer(modifier = Modifier.width(4.dp))
                                    Text("Dispose / Remove", fontSize = 12.sp, color = AlertRed, fontWeight = FontWeight.Bold)
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}

File: lib/screens/expiry_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../core/theme/app_colors.dart';
import '../providers/pharmacy_provider.dart';
import '../routes/app_routes.dart';
import '../widgets/medicine/medicine_card.dart';

class ExpiryScreen extends StatefulWidget {
  const ExpiryScreen({super.key});

  @override
  State<ExpiryScreen> createState() => _ExpiryScreenState();
}

class _ExpiryScreenState extends State<ExpiryScreen> with SingleTickerProviderStateMixin {
  late TabController _tabController;

  @override
  void initState() {
    super.initState();
    _tabController = TabController(length: 2, vsync: this);
    _tabController.addListener(() {
      setState(() {});
    });
  }

  @override
  void dispose() {
    _tabController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<PharmacyProvider>();
    final expiredList = provider.expiredMedicines;
    final expiringSoonList = provider.expiringSoonMedicines;

    return Scaffold(
      backgroundColor: AppColors.surfaceWhite,
      appBar: AppBar(
        backgroundColor: AppColors.surfaceWhite,
        elevation: 0,
        title: const Text(
          "Expiry Management",
          style: TextStyle(fontWeight: FontWeight.bold, color: AppColors.textPrimary),
        ),
        leading: IconButton(
          icon: const Icon(Icons.arrow_back, color: AppColors.textPrimary),
          onPressed: () => Navigator.pop(context),
        ),
      ),
      body: Column(
        children: [
          const SizedBox(height: 12),
          Padding(
            padding: const EdgeInsets.symmetric(horizontal: 16.0),
            child: Container(
              decoration: BoxDecoration(
                color: AppColors.surfaceCard,
                borderRadius: BorderRadius.circular(12),
              ),
              child: TabBar(
                controller: _tabController,
                indicatorColor: _tabController.index == 0 ? AppColors.alertRed : AppColors.warningOrange,
                indicatorWeight: 3,
                labelColor: _tabController.index == 0 ? AppColors.alertRed : AppColors.warningOrange,
                unselectedLabelColor: AppColors.textSecondary,
                labelStyle: const TextStyle(fontWeight: FontWeight.bold),
                tabs: [
                  Tab(text: "Expired (${expiredList.length})"),
                  Tab(text: "Expiring in 30 Days (${expiringSoonList.length})"),
                ],
              ),
            ),
          ),
          const SizedBox(height: 16),
          Expanded(
            child: TabBarView(
              controller: _tabController,
              children: [
                _buildExpiryList(context, expiredList, true),
                _buildExpiryList(context, expiringSoonList, false),
              ],
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildExpiryList(BuildContext context, List<dynamic> list, bool isExpiredTab) {
    if (list.isEmpty) {
      return Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            const Icon(
              Icons.warning_rounded,
              size: 48,
              color: AppColors.successGreen,
            ),
            const SizedBox(height: 12),
            Text(
              isExpiredTab ? "No Expired Medicines!" : "No Medicines Expiring in 30 Days!",
              style: const TextStyle(
                fontSize: 16,
                fontWeight: FontWeight.bold,
                color: AppColors.textPrimary,
              ),
            ),
            const Text(
              "Your inventory is fresh and safe.",
              style: TextStyle(fontSize: 13, color: AppColors.textSecondary),
            ),
          ],
        ),
      );
    }

    return ListView.separated(
      padding: const EdgeInsets.fromLTRB(16, 0, 16, 24),
      itemCount: list.length,
      separatorBuilder: (context, index) => const SizedBox(height: 12),
      itemBuilder: (context, index) {
        final medicine = list[index];
        return Column(
          children: [
            MedicineCard(
              medicine: medicine,
              isExpired: isExpiredTab,
              isExpiringSoon: !isExpiredTab,
              onTap: () => Navigator.pushNamed(
                context,
                AppRoutes.productDetails,
                arguments: medicine.id,
              ),
            ),
            const SizedBox(height: 6),
            Row(
              mainAxisAlignment: MainAxisAlignment.end,
              children: [
                if (!isExpiredTab)
                  TextButton.icon(
                    onPressed: () {
                      ScaffoldMessenger.of(context).showSnackBar(
                        SnackBar(content: Text("Clearance 30% discount tag added to ${medicine.name}")),
                      );
                    },
                    icon: const Icon(Icons.discount, size: 16, color: AppColors.warningOrange),
                    label: const Text(
                      "Apply Clearance Discount",
                      style: TextStyle(fontSize: 12, color: AppColors.warningOrange, fontWeight: FontWeight.bold),
                    ),
                  ),
                TextButton.icon(
                  onPressed: () {
                    context.read<PharmacyProvider>().deleteMedicine(medicine.id);
                    ScaffoldMessenger.of(context).showSnackBar(
                      SnackBar(content: Text("${medicine.name} removed from inventory")),
                    );
                  },
                  icon: const Icon(Icons.delete, size: 16, color: AppColors.alertRed),
                  label: const Text(
                    "Dispose / Remove",
                    style: TextStyle(fontSize: 12, color: AppColors.alertRed, fontWeight: FontWeight.bold),
                  ),
                ),
              ],
            ),
          ],
        );
      },
    );
  }
}
